### PR TITLE
E2e appium/send native and erc tokens

### PR DESCRIPTION
**Pull request by Andepande**
**Description**
- SendToken Flow originating from this [ticket](https://github.com/MetaMask/mobile-planning/issues/546)
- Increased Appium test coverage 
- Minor implementations to improve test runs

**ScreenShots**
image

Failing tests
/wdio/features/ExploringWizard.feature - Passed locally
/wdio/features/BrowserFlow/ENSWebsite.feature - Last step Browser tests currently on pause
/wdio/features/AppLaunchTime.feature - Fails due to biometric screen( Passes when run in Isolation
/wdio/features/AddressFlow.feature - Fails due to no Eth on test account
/wdio/features/ImportCustomToken.feature Fails on my machine due to lack of importing Goerli token, passes on other machines

**Issue**

Progresses #???

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/Views/SendFlow/Amount/index.js [hunk 1]
```diff
@@ -1120,7 +1121,7 @@ class Amount extends PureComponent {
               keyboardType={'numeric'}
               placeholder={'0'}
               placeholderTextColor={colors.text.muted}
-              testID={'txn-amount-input'}
+              {...generateTestId(Platform, TRANSACTION_INPUT_ID)}
```
Comment (cortisiko) [inline app/components/Views/SendFlow/Amount/index.js:1124]: update detox. In particular, ensure that https://github.com/MetaMask/metamask-mobile/blob/main/e2e/pages/AmountView.js#L3 imports the element locator ID from the wdio folder

File: app/components/Views/SendFlow/Confirm/index.js [hunk 2]
```diff
@@ -1221,7 +1224,10 @@ class Confirm extends PureComponent {
               <Text style={styles.textAmountLabel}>
                 {strings('transaction.amount')}
               </Text>
-              <Text style={styles.textAmount} testID={'confirm-txn-amount'}>
+              <Text
+                style={styles.textAmount}
+                {...generateTestId(Platform, COMFIRM_TXN_AMOUNT)}
```
Comment (cortisiko) [inline app/components/Views/SendFlow/Confirm/index.js:1229]: update detox. In particular, ensure thathttps://github.com/MetaMask/metamask-mobile/blob/main/e2e/pages/TransactionConfirmView.js#L4 imports the element locator ID from the wdio folder

File: wdio/screen-objects/SendScreen.js [hunk 3]
```diff
@@ -7,6 +7,8 @@ import {
   AMOUNT_SCREEN,
   ADD_ADDRESS_BUTTON,
   SEND_CANCEL_BUTTON,
+  TRANSACTION_INPUT_ID,
+  COMFIRM_TXN_AMOUNT,
 } from './testIDs/Screens/SendScreen.testIds';
 
 class SendScreen {
```
Comment (cortisiko) [inline wdio/screen-objects/SendScreen.js:14]: I am in favor of creating two more page objects out of this class. One for the amount screen and another for the Transaction confirmation screen. 

The amount screen should have these methods:
```
amountInputField(),
confirmAmount(),
enterAmount(),
```
The transaction confirmation screen should have these methods:
```
isConfirmScreenVisible()
isCorrectTokenConfirm()
isCorrectTokenAmountDisplayed()
```

File: wdio/screen-objects/TokenScreen.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import {
+  TOKEN_ASSET_OVERVIEW,
+  ASSET_BACK_BUTTON,
+} from '../screen-objects/testIDs/Screens/TokenOverviewScreen.testIds';
+import Selectors from '../helpers/Selectors';
+import Gestures from '../helpers/Gestures';
+
+class TokenScreen {
```
Comment (cortisiko) [inline wdio/screen-objects/TokenScreen.js:8]: quick question, which screen is this one? is it the amount screen?

**Review by cortisiko — CHANGES_REQUESTED** [review]
left a few comments! Run detox to ensure nothing broke.

File: wdio/screen-objects/Onboarding/OnboardingCarousel.js [hunk 5]
```diff
@@ -28,10 +28,11 @@ class WelcomeScreen {
...
-    await elem.waitForDisplayed();
+    await elem.waitForExist({ reverse: false });
+    await elem.waitForExist({ reverse: true });
   }
 
-  async waitForSplashAnimationToNotExit() {
+  async waitForSplashAnimationToNotExist() {
     const elem = await this.splashScreenMetamaskAnimationId;
     await elem.waitForExist({ reverse: true });
   }
```
Comment (SamuelSalas) [inline wdio/screen-objects/Onboarding/OnboardingCarousel.js:38]: This change might affect the app launch time test scenario. The timer should stop in the exactly after the splash animation has been displayed. So, for not duplicating code, it was defined to move the displayed or waitforexist method to another step definition and add the waitforexist reverse in the original step.

File: wdio/screen-objects/TokenScreen.js (hunk 4 above)
Comment (Andepande) [reply wdio/screen-objects/TokenScreen.js:8]: This is the Token overview screen

File: wdio/screen-objects/Onboarding/OnboardingCarousel.js (hunk 5 above)
Comment (Andepande) [reply wdio/screen-objects/Onboarding/OnboardingCarousel.js:38]: Can we have a call on this when you log on?

**Pull request closed without merging**